File: C++ Kasus/src/Keyboard.cpp
#include "myclass/Keyboard.h"
#include <iostream>
#include <string>

std::string Keyboard::Display()
{
    return "ini keyboard";
}

Keyboard::Keyboard(std::string inputNama, int inputHarga, int inputStok, std::string backlight, int banyakKey, std::string tipeSwitch) : Barang(inputNama, inputHarga, inputStok)
{

    this->backlight = backlight;
    this->banyakKey = banyakKey;
    this->tipeSwitch = tipeSwitch;
}

void Keyboard::infoBarang()
{
    Barang::infoBarang();
    std::cout << " pcs., Backlight : " << this->backlight << ", Banyak Keys : " << this->banyakKey << ", Tipe Switch : " << this->tipeSwitch << "\n"
              << std::endl;
}

void Keyboard::setBacklight(std::string backlight)
{
    this->backlight = backlight;
}

void Keyboard::setBanyakKey(int banyakKey)
{
    this->banyakKey = banyakKey;
}

void Keyboard::setTipeSwitch(std::string tipeSwitch)
{
    this->tipeSwitch = tipeSwitch;
}

std::string Keyboard::getBacklight()
{
    return this->backlight;
}

int Keyboard::getBanyakKey()
{
    return this->banyakKey;
}

std::string Keyboard::getTipeSwitch()
{
    return this->tipeSwitch;
}

File: Java/Java Kasus/src/com/myclass/Keyboard.java
package com.myclass;

public class Keyboard extends Barang {
    private String backlight, tipeSwitch;
    private int banyakKeys;

    public String displayBarang() {
        return "ini Keyboard";
    }

    public Keyboard(String namaBarang, int hargaBarang, int stokBarang, String backlight, int banyakKeys,
            String tipeSwitch) {
        super(namaBarang, hargaBarang, stokBarang);
        this.backlight = backlight;
        this.banyakKeys = banyakKeys;
        this.tipeSwitch = tipeSwitch;
    }

    public void setBacklight(String backlight) {
        this.backlight = backlight;
    }

    public void setBanyakKeys(int banyakKeys) {
        this.banyakKeys = banyakKeys;
    }

    public void setTipeSwitch(String tipeSwitch) {
        this.tipeSwitch = tipeSwitch;
    }

    public String getBacklight() {
        return this.backlight;
    }

    public int getBanyakKeys() {
        return this.banyakKeys;
    }

    public String getTipeSwitch() {
        return this.tipeSwitch;
    }

    @Override
    public void infoBarang() {
        super.infoBarang();
        System.out.println("Backlight : " + this.backlight + ", Banyak keys : " + this.banyakKeys + ", Tipe Switch : "
                + this.tipeSwitch + "\n");
    }
}
